File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l = 0
        r = 0
        maxProfit = 0
        while r != len(prices):
            if prices[l] < prices[r]:
                maxProfit = max(prices[r]-prices[l], maxProfit)
            if prices[l] >= prices[r]:
                l = r
            r+=1
        return maxProfit

File: Data Structures & Algorithms/max-water-container/submission-1.java
class Solution {
    public int maxArea(int[] heights) {
     int l = 0;
     int r = heights.length - 1;
     int maxwater = 0;

        while (l < r)
        {
            if(heights[l] > heights[r])
            {
                maxwater = Math.max(maxwater, (r-l)*heights[r]);
                r--;
            }
            else if (heights[l] <= heights[r])
            {
                maxwater = Math.max(maxwater, (r-l)*heights[l]);
                l++;
            }
        }
        return maxwater;
    }
}
